zeitgeist-sample.js: NODE_PATH not resolved when run from skill directory

*Filed by Muninn*

## Bug

`zeitgeist-sample.js` uses `require("ws")` and `require("https-proxy-agent")`, which resolve relative to the script location (`/mnt/skills/user/sampling-bluesky-zeitgeist/scripts/`). The skill instructs users to `npm install` in `/home/claude`, but Node does not search that `node_modules` directory when the script lives elsewhere.

### Current behavior
```
Error: Cannot find module 'ws'
Require stack:
- /mnt/skills/user/sampling-bluesky-zeitgeist/scripts/zeitgeist-sample.js
```

### Workaround
```bash
export NODE_PATH=/home/claude/node_modules
```

### Fix options (pick one)
1. **In the script**: Add `module.paths.unshift("/home/claude/node_modules")` before any requires
2. **In SKILL.md**: Add `export NODE_PATH=/home/claude/node_modules` to the setup instructions
3. **Both**: Belt and suspenders

Option 1 is cleanest — makes the skill self-contained regardless of invocation context.

--- a/browsing-bluesky/CHANGELOG.md
+++ b/browsing-bluesky/CHANGELOG.md
@@ -2,6 +2,18 @@
 
 All notable changes to the `browsing-bluesky` skill are documented in this file. The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
+## [0.5.0] - 2026-02-08
+
+### Added
+
+- Add `get_trending()` for rich trend data with post counts, status, category, and actors (#272)
+- Add `get_trending_topics()` for lightweight trending topic scan (#272)
+- Restructure zeitgeist workflow: trending API first, firehose as deep-dive tool (#272)
+
+### Fixed
+
+- Fix NODE_PATH resolution in zeitgeist-sample.js so modules resolve regardless of working directory (#271)
+
 ## [0.4.1] - 2026-02-06
 
 ### Added

--- a/browsing-bluesky/README.md
+++ b/browsing-bluesky/README.md
@@ -163,4 +163,4 @@ This skill consolidates and replaces:
 
 ## Version
 
-0.1.0
+0.5.0

--- a/browsing-bluesky/SKILL.md
+++ b/browsing-bluesky/SKILL.md
@@ -2,7 +2,7 @@
 name: browsing-bluesky
 description: Browse Bluesky content via API and firehose - search posts, fetch user activity, sample trending topics, read feeds and lists, analyze and categorize accounts. Supports authenticated access for personalized feeds. Use for Bluesky research, user monitoring, trend analysis, feed reading, firehose sampling, account categorization.
 metadata:
-  version: 0.4.1
+  version: 0.5.0
 ---
 
 # Browsing Bluesky
@@ -21,6 +21,8 @@ from browsing_bluesky import (
     search_posts, get_user_posts, get_profile, get_feed_posts, sample_firehose,
     get_thread, get_quotes, get_likes, get_reposts,
     get_followers, get_following, search_users,
+    # Trending
+    get_trending, get_trending_topics,
     # Account analysis
     get_all_following, get_all_followers, extract_post_text,
     extract_keywords, analyze_account, analyze_accounts,
@@ -80,22 +82,47 @@ Combine query syntax with function params for complex searches.
 2. Get recent posts with `get_user_posts(handle, limit=N)`
 3. For topic-specific user content, use `search_posts(query, author=handle)`
 
-### Sample Trending Topics
+### Discover What's Trending
+
+**Recommended workflow** — trending API first, firehose for deep dives:
+
+#### 1. Quick scan with trending topics (~500 tokens)
+
+```python
+topics = get_trending_topics(limit=10)
+# Returns: {topics: [{topic, display_name, description, link}, ...],
+#           suggested: [...]}
+```
+
+#### 2. Rich trends with post counts and actors
+
+```python
+trends = get_trending(limit=10)
+for t in trends:
+    print(f"{t['display_name']} — {t['post_count']} posts ({t['status']})")
+# Each trend includes: topic, display_name, link, started_at,
+#   post_count, status, category, actors
+```
+
+#### 3. Targeted exploration of selected trends
+
+```python
+posts = search_posts(trend["topic"], limit=25)
+```
+
+#### 4. Optional: Firehose for velocity monitoring or long-tail discovery
 
 **Prerequisites**: Install Node.js dependencies once per session:
 ```bash
 cd /home/claude && npm install ws https-proxy-agent 2>/dev/null
 ```
 
-**Usage**:
 ```python
-data = sample_firehose(duration=30)  # Sample for 30 seconds
-data = sample_firehose(duration=20, filter="python")  # Filter for specific term
+data = sample_firehose(duration=30)  # Full firehose sample
+data = sample_firehose(duration=20, filter="python")  # Filtered sample
 ```
 
 Returns: `topWords`, `topPhrases`, `entities`, `samplePosts`, `stats`
-
-Use for real-time zeitgeist, trending topic detection, or filtered stream analysis.
 
 ### Read Feeds and Lists
 
@@ -162,6 +189,7 @@ for u in users:
 
 - **Public AppView**: `https://api.bsky.app/xrpc/` for unauthenticated reads
 - **PDS**: `https://bsky.social/xrpc/` for authenticated requests
+- **Trending**: `app.bsky.unspecced.getTrends` (rich) and `app.bsky.unspecced.getTrendingTopics` (lightweight)
 - **Firehose**: `wss://jetstream1.us-east.bsky.network/subscribe`
 - **Endpoint routing** is automatic - authenticated requests go to PDS, public requests go to AppView
 - **Rate limits** exist but are generous for read operations

--- a/browsing-bluesky/_MAP.md
+++ b/browsing-bluesky/_MAP.md
@@ -9,11 +9,13 @@
 
 ### CHANGELOG.md
 - browsing-bluesky - Changelog `h1` :1
-- [0.4.0] - 2026-01-25 `h2` :5
-- [0.4.0] - 2026-01-25 `h2` :15
-- [0.3.0] - 2026-01-25 `h2` :28
-- [0.2.0] - 2026-01-01 `h2` :41
-- [0.1.0] - 2026-01-01 `h2` :51
+- [0.5.0] - 2026-02-08 `h2` :5
+- [0.4.1] - 2026-02-06 `h2` :17
+- [0.4.0] - 2026-01-25 `h2` :27
+- [0.4.0] - 2026-01-25 `h2` :37
+- [0.3.0] - 2026-01-25 `h2` :50
+- [0.2.0] - 2026-01-01 `h2` :63
+- [0.1.0] - 2026-01-01 `h2` :73
 
 ### README.md
 - browsing-bluesky `h1` :1
@@ -30,11 +32,11 @@
 ### SKILL.md
 - Browsing Bluesky `h1` :8
 - Implementation `h2` :12
-- Authentication (Optional) `h2` :32
-- Research Workflows `h2` :64
-- API Endpoint Notes `h2` :161
-- Return Format `h2` :169
-- Account Analysis `h2` :184
+- Authentication (Optional) `h2` :34
+- Research Workflows `h2` :66
+- API Endpoint Notes `h2` :188
+- Return Format `h2` :197
+- Account Analysis `h2` :212
 
 ### __init__.py
 > Imports: `.scripts.bsky`

--- a/browsing-bluesky/__init__.py
+++ b/browsing-bluesky/__init__.py
@@ -14,6 +14,9 @@ from .scripts.bsky import (
     get_followers,
     get_following,
     search_users,
+    # Trending
+    get_trending,
+    get_trending_topics,
     # Account analysis (from categorizing-bsky-accounts)
     get_all_following,
     get_all_followers,
@@ -41,6 +44,9 @@ __all__ = [
     "get_followers",
     "get_following",
     "search_users",
+    # Trending
+    "get_trending",
+    "get_trending_topics",
     # Account analysis
     "get_all_following",
     "get_all_followers",

--- a/browsing-bluesky/scripts/_MAP.md
+++ b/browsing-bluesky/scripts/_MAP.md
@@ -23,27 +23,29 @@
     limit: int = 25
 )` :216
 - **get_feed_posts** (f) `(feed_uri: str, limit: int = 20)` :259
-- **sample_firehose** (f) `(duration: int = 10, filter: Optional[str] = None)` :301
-- **get_thread** (f) `(post_uri_or_url: str, depth: int = 6, parent_height: int = 80)` :333
-- **get_quotes** (f) `(post_uri_or_url: str, limit: int = 25)` :354
-- **get_likes** (f) `(post_uri_or_url: str, limit: int = 50)` :373
-- **get_reposts** (f) `(post_uri_or_url: str, limit: int = 50)` :392
-- **get_followers** (f) `(handle: str, limit: int = 50)` :411
-- **get_following** (f) `(handle: str, limit: int = 50)` :430
-- **search_users** (f) `(query: str, limit: int = 25)` :449
-- **get_all_following** (f) `(handle: str, limit: int = 100)` :516
-- **get_all_followers** (f) `(handle: str, limit: int = 100)` :537
-- **extract_post_text** (f) `(posts: List[Dict[str, Any]])` :558
+- **get_trending** (f) `(limit: int = 10)` :301
+- **get_trending_topics** (f) `(limit: int = 10)` :337
+- **sample_firehose** (f) `(duration: int = 10, filter: Optional[str] = None)` :374
+- **get_thread** (f) `(post_uri_or_url: str, depth: int = 6, parent_height: int = 80)` :406
+- **get_quotes** (f) `(post_uri_or_url: str, limit: int = 25)` :427
+- **get_likes** (f) `(post_uri_or_url: str, limit: int = 50)` :446
+- **get_reposts** (f) `(post_uri_or_url: str, limit: int = 50)` :465
+- **get_followers** (f) `(handle: str, limit: int = 50)` :484
+- **get_following** (f) `(handle: str, limit: int = 50)` :503
+- **search_users** (f) `(query: str, limit: int = 25)` :522
+- **get_all_following** (f) `(handle: str, limit: int = 100)` :589
+- **get_all_followers** (f) `(handle: str, limit: int = 100)` :610
+- **extract_post_text** (f) `(posts: List[Dict[str, Any]])` :631
 - **extract_keywords** (f) `(
     text: str,
     top_n: int = 10,
     stopwords: str = "en"
-)` :570
+)` :643
 - **analyze_account** (f) `(
     handle: str,
     posts_limit: int = 20,
     stopwords: str = "en"
-)` :662
+)` :735
 - **analyze_accounts** (f) `(
     handles: Optional[List[str]] = None,
     following: Optional[str] = None,
@@ -52,7 +54,7 @@
     posts_per_account: int = 20,
     stopwords: str = "en",
     exclude_patterns: Optional[List[str]] = None
-)` :700
+)` :773
 
 ### zeitgeist-sample.js
 - *No top-level symbols*

--- a/browsing-bluesky/scripts/__init__.py
+++ b/browsing-bluesky/scripts/__init__.py
@@ -1,11 +1,29 @@
-"""Bluesky browsing module - API and firehose access."""
+"""Bluesky browsing module - API, trending, and firehose access."""
 
 from .bsky import (
     search_posts,
     get_user_posts,
     get_profile,
     get_feed_posts,
-    sample_firehose
+    sample_firehose,
+    get_thread,
+    get_quotes,
+    get_likes,
+    get_reposts,
+    get_followers,
+    get_following,
+    search_users,
+    get_trending,
+    get_trending_topics,
+    get_all_following,
+    get_all_followers,
+    extract_post_text,
+    extract_keywords,
+    analyze_account,
+    analyze_accounts,
+    is_authenticated,
+    get_authenticated_user,
+    clear_session
 )
 
 __all__ = [
@@ -13,5 +31,23 @@ __all__ = [
     "get_user_posts",
     "get_profile",
     "get_feed_posts",
-    "sample_firehose"
+    "sample_firehose",
+    "get_thread",
+    "get_quotes",
+    "get_likes",
+    "get_reposts",
+    "get_followers",
+    "get_following",
+    "search_users",
+    "get_trending",
+    "get_trending_topics",
+    "get_all_following",
+    "get_all_followers",
+    "extract_post_text",
+    "extract_keywords",
+    "analyze_account",
+    "analyze_accounts",
+    "is_authenticated",
+    "get_authenticated_user",
+    "clear_session"
 ]

--- a/browsing-bluesky/scripts/bsky.py
+++ b/browsing-bluesky/scripts/bsky.py
@@ -298,6 +298,79 @@ def get_feed_posts(feed_uri: str, limit: int = 20) -> List[Dict[str, Any]]:
     return [_parse_post(item["post"]) for item in r.json().get("feed", [])]
 
 
+def get_trending(limit: int = 10) -> List[Dict[str, Any]]:
+    """Get current trending topics with post counts and top actors.
+
+    Uses app.bsky.unspecced.getTrends for rich trend data including
+    post counts, status, category, and participating actors.
+
+    Args:
+        limit: Max trends to return (default 10, max 25)
+
+    Returns:
+        List of trend dicts with topic, displayName, link, startedAt,
+        postCount, status, category, actors
+    """
+    base, headers = _get_base_and_headers()
+    r = requests.get(
+        f"{base}/app.bsky.unspecced.getTrends",
+        params={"limit": min(limit, 25)},
+        headers=headers
+    )
+    r.raise_for_status()
+    trends = r.json().get("trends", [])
+    results = []
+    for t in trends:
+        results.append({
+            "topic": t.get("topic"),
+            "display_name": t.get("displayName"),
+            "link": t.get("link"),
+            "started_at": t.get("startedAt"),
+            "post_count": t.get("postCount", 0),
+            "status": t.get("status"),
+            "category": t.get("category"),
+            "actors": [_parse_actor(a) for a in t.get("actors", [])],
+        })
+    return results
+
+
+def get_trending_topics(limit: int = 10) -> Dict[str, Any]:
+    """Get lightweight list of trending topics and suggestions.
+
+    Uses app.bsky.unspecced.getTrendingTopics for a compact overview.
+    Lower token cost than get_trending() — use this for a quick scan
+    before deciding which topics to explore further.
+
+    Args:
+        limit: Max topics to return (default 10, max 25)
+
+    Returns:
+        Dict with 'topics' and 'suggested' lists, each containing
+        dicts with topic, displayName, description, link
+    """
+    base, headers = _get_base_and_headers()
+    r = requests.get(
+        f"{base}/app.bsky.unspecced.getTrendingTopics",
+        params={"limit": min(limit, 25)},
+        headers=headers
+    )
+    r.raise_for_status()
+    data = r.json()
+
+    def _parse_topic(t: Dict) -> Dict[str, Any]:
+        return {
+            "topic": t.get("topic"),
+            "display_name": t.get("displayName"),
+            "description": t.get("description"),
+            "link": t.get("link"),
+        }
+
+    return {
+        "topics": [_parse_topic(t) for t in data.get("topics", [])],
+        "suggested": [_parse_topic(t) for t in data.get("suggested", [])],
+    }
+
+
 def sample_firehose(duration: int = 10, filter: Optional[str] = None) -> Dict[str, Any]:
     """Sample the Bluesky firehose for trending topics.
 

--- a/browsing-bluesky/scripts/zeitgeist-sample.js
+++ b/browsing-bluesky/scripts/zeitgeist-sample.js
@@ -4,6 +4,12 @@
  * Samples firehose for N seconds, outputs structured JSON for analysis
  */
 
+// Ensure node_modules installed in home directory are resolvable regardless
+// of the working directory this script is invoked from (fixes #271)
+const path = require('path');
+const homeDir = process.env.HOME || '/home/claude';
+module.paths.unshift(path.join(homeDir, 'node_modules'));
+
 const WebSocket = require('ws');
 const { HttpsProxyAgent } = require('https-proxy-agent');
 

--- a/sampling-bluesky-zeitgeist/scripts/zeitgeist-sample.js
+++ b/sampling-bluesky-zeitgeist/scripts/zeitgeist-sample.js
@@ -4,6 +4,12 @@
  * Samples firehose for N seconds, outputs structured JSON for analysis
  */
 
+// Ensure node_modules installed in home directory are resolvable regardless
+// of the working directory this script is invoked from (fixes #271)
+const path = require('path');
+const homeDir = process.env.HOME || '/home/claude';
+module.paths.unshift(path.join(homeDir, 'node_modules'));
+
 const WebSocket = require('ws');
 const { HttpsProxyAgent } = require('https-proxy-agent');
 

